### ISSUE TITLE
Browserify/babelify transforms not working

### ISSUE THREAD

**Issue by jamby** [browserify-rails/browserify-rails#181]
So I'm trying to get 2 plugins working with my browserify/babelify.

I've got inside `devDependencies`:

``` javascript
"babel-plugin-transform-class-properties": "^6.18.0",
"babel-plugin-transform-decorators-legacy": "^1.3.4",
"babel-preset-es2015": "^6.18.0",
"babel-preset-react": "^6.16.0",
"babel-preset-stage-0": "^6.16.0",
```

And then inside my `application.rb`:

``` ruby
config.browserify_rails.commandline_options = '-t [ babelify --presets [ es2015 stage-0 react ] --plugins [ transform-decorators-legacy transform-class-properties ] --extensions [.jsx .jsx.erb ] ]'
```

But when I attempt to do `@autobind` or the `static propTypes = {`, I still get a `ActionView::Template::Error (SyntaxError: unknown: Unexpected token`.

Was wondering if anyone else has tried to do class properties or decorators with their transforms.

Is there something I'm missing that's not allowing this to work?

**Comment by cguillemette** [comment]
@jamby Never used @autobind before but wouldn't you need the plugin (class-autobind)?

**Comment by rpearce** [comment]
@jamby I believe there is an issue with what code gets loaded first. Try doing these things:

1. Make sure your `application.js` requires in what you need and includes nothing but a couple of `require` statements / valid ES5 code
2. Move your transform command line options to your `package.json` kind of like so
```js
  "browserify": {
    "transform": [
      [
        "babelify"
      ]
    ]
  }
```
3. move your presets to a `.babelrc` file
```js
{
  "presets": ["es2015", "react"]
}
```

**Comment by jamby** [comment]
Yeah @rpearce we actually figured it out awhile ago that trying to add all those options to the `commandline_options` was not working but making a `.babelrc` file did work.

However, then we ran into issues with needing to do things with `.erb` files. We've kinda curbed the adding of babelify and yarn.

However, it seems Rails 5 is adding in `--yarn` config into 5.1, and soon after will maybe have --`webpack` config. See: https://github.com/rails/rails/pull/26836#issuecomment-264242030

So maybe in the future this won't even be needed.
